# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/mac/mozActionElements.mm
# Commit: dcc478201cb7
# Full Hash: dcc478201cb75d078d3a5fd33f306c7b9cfcd527
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2021-08-27 03:42:22
# Regressor Bug: 1727643
# File Overlap Count: 1
# Description:
#   Bug 1727643 - P4: Make use of new common methods in platforms and XPCOM. r=morgan
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123698
# ==============================================================================

diff -r 1a9901750064 -r dcc478201cb7 accessible/mac/mozActionElements.mm
--- a/accessible/mac/mozActionElements.mm	Thu Aug 26 19:40:05 2021 +0000
+++ b/accessible/mac/mozActionElements.mm	Thu Aug 26 19:40:06 2021 +0000
@@ -188,14 +188,9 @@
 - (void)changeValueBySteps:(int)factor {
   MOZ_ASSERT(mGeckoAccessible, "mGeckoAccessible is null");
 
-  if (LocalAccessible* acc = mGeckoAccessible->AsLocal()) {
-    double newValue = acc->CurValue() + (acc->Step() * factor);
-    [self setValue:(newValue)];
-  } else {
-    RemoteAccessible* proxy = mGeckoAccessible->AsRemote();
-    double newValue = proxy->CurValue() + (proxy->Step() * factor);
-    [self setValue:(newValue)];
-  }
+  double newValue =
+      mGeckoAccessible->CurValue() + (mGeckoAccessible->Step() * factor);
+  [self setValue:(newValue)];
 }
 
 /*
@@ -204,22 +199,14 @@
 - (void)setValue:(double)value {
   MOZ_ASSERT(mGeckoAccessible, "mGeckoAccessible is null");
 
-  if (LocalAccessible* acc = mGeckoAccessible->AsLocal()) {
-    double min = acc->MinValue();
-    double max = acc->MaxValue();
-    // Because min and max are not required attributes, we first check
-    // if the value is undefined. If this check fails,
-    // the value is defined, and we verify our new value falls
-    // within the bound (inclusive).
-    if ((IsNaN(min) || value >= min) && (IsNaN(max) || value <= max)) {
+  double min = mGeckoAccessible->MinValue();
+  double max = mGeckoAccessible->MaxValue();
+
+  if ((IsNaN(min) || value >= min) && (IsNaN(max) || value <= max)) {
+    if (LocalAccessible* acc = mGeckoAccessible->AsLocal()) {
       acc->SetCurValue(value);
-    }
-  } else {
-    RemoteAccessible* proxy = mGeckoAccessible->AsRemote();
-    double min = proxy->MinValue();
-    double max = proxy->MaxValue();
-    // As above, check if the value is within bounds.
-    if ((IsNaN(min) || value >= min) && (IsNaN(max) || value <= max)) {
+    } else {
+      RemoteAccessible* proxy = mGeckoAccessible->AsRemote();
       proxy->SetCurValue(value);
     }
   }